forem/forem
# Wellness badge query+service refactor

## What type of PR is this? (check all applicable)


## Description

Based on executions over the past week this PR refactors the query and the service that awards the badges.

## Related Tickets & Documents



- Related to #17310

## QA Instructions, Screenshots, Recordings

Just as previous PRs this one is tricky to QA locally. I'd recommend reading through both specs related to this PR and see if the logic makes sense to you.

If you want to QA this locally take a look at the [instructions from this PR](https://github.com/forem/forem/pull/17394#issue-1211273307)

### UI accessibility concerns?

None

## Added/updated tests?


## [Forem core team only] How will this change be communicated?


## [optional] Are there any post deployment tasks we need to perform?

Check the upcoming dry-run results and compare them with previous runs

## [optional] What gif best describes this PR or how it makes you feel?

![knobs being adjusted - tweak](https://media.giphy.com/media/kiPrADvAjcZeKLnZkv/giphy.gif)
